### ISSUE TITLE
Error when importing sionna on Apple M1 Max

### ISSUE THREAD

**Issue by krono-i2** [NVlabs/sionna#242]
Hello everybody,
I'm trying to use the sionna framework on Apple M1 Max SoC.
Tensorflow and Keras work like a charm on GPU.
The problem is that it seems the ray tracer included in sionna automatically tries to use the variant `cuda_ad_rgb` of mitsuba renderer when a GPU is found by tensorflow.
When I try to import sionna I obtain:
```
>>> import sionna
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/krono/mambaforge/envs/sionna/lib/python3.11/site-packages/sionna/__init__.py", line 19, in <module>
    from . import rt
  File "/Users/krono/mambaforge/envs/sionna/lib/python3.11/site-packages/sionna/rt/__init__.py", line 22, in <module>
    mi.set_variant('cuda_ad_rgb')
  File "/Users/krono/mambaforge/envs/sionna/lib/python3.11/site-packages/mitsuba/__init__.py", line 300, in set_variant
    raise ImportError('Requested an unsupported variant "%s". The '
ImportError: Requested an unsupported variant "cuda_ad_rgb". The following variants are available: scalar_rgb, scalar_spectral, llvm_ad_rgb.
>>> 
```
Observing the file `/Users/krono/mambaforge/envs/sionna/lib/python3.11/site-packages/sionna/rt/__init__.py` I've found the following lines.
```
gpus = tf.config.list_physical_devices('GPU')
if len(gpus) > 0:
    mi.set_variant('cuda_ad_rgb')
else:
    mi.set_variant('llvm_ad_rgb')
```
As you can see, when a GPU is found sionna automatically tries to set the variant `cuda_ad_rgb`.
I found out mitsuba does not support the Apple Silicon GPUs, so we have two alternatives:
- use a renderer that support Apple Metal (e.g., Octane X);
- use a CPU renderer on Apple Silicon SoC (much slower).

How can I force, at least, to use `llvm_ad_rgb`, even if a GPU is found?
Thanks for help.

Ivan Iudice

**Comment by merlinND** [comment]
Hello @krono-i2,

You are correct that there is no Metal support in Mitsuba, so you would have to use the CPU (i.e. LLVM backend).
Without modifying the Sionna code, according to [this post](https://developer.apple.com/forums/thread/708999), you can hide GPU devices from TensorFlow by adding the following line right after importing Tensorflow:

```
tf.config.set_visible_devices([], 'GPU')
```

**Comment by krono-i2** [comment]
Hello @merlinND,
thanks for answer!
The docker installation is still the preferred way? This could solve the issue automatically.

**Comment by merlinND** [comment]
The Docker image is definitely a convenient way to get started, however either methods should work fine.

**Comment by krono-i2** [comment]
Looking at the DOCKERFILE I've seen the used image is `tensorflow/tensorflow:2.11.0-gpu-jupyter`. Obviously, it is configured for platform linux/amd64.
I've found out that all the tensorflow images provided on dockerhub are configured for platform linux/amd64.
So, I think is not straight forward to install sionna via docker on Apple Silicon.
I tried it using image `tensorflow/tensorflow:2.11.0-jupyter`, but ipython kernel simply stucks when importing sionna.
